.github/workflows/typescript-workflow.yml

```diff
@@ -0,0 +1,20 @@
+name: 'Type checking for typescript'
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  test:
+    name: 'Validating Type'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install pnpm
+        run: npm install -g pnpm
+
+      - name: Install dependencies
+        run: pnpm install
```

Comment:
I want to remove repetitively installing pnpm, dependencies in https://github.com/suspensive/react/pull/20/files#diff-e894bfa63f52fe487b10fce0d90f8276523f2a08f5120025aaafe3e5c72a5d23R13-R17, How about merge this file with eslint-workflow.yml in https://github.com/suspensive/react/pull/20 and rename this file like `ci-workflow.yml`? and add workflow's job for testing code by jest after?

I think it would be better if we test all(type:check, test by jest, eslint) in ci-workflow.yml at once, with allowing only one time to install pnpm, dependencies in ci-workflow. How do you think this?

Reply:
I thought it would be better to divide it to make sure that we know where we failed.
After read your comment, I think it's better to run it in the same workflow for performance!
Also, if we go into the [action page](https://github.com/suspensive/react/actions), We will see where our code failed in detail, so it would be good to merge with the same file! I'll fix it😃